embassy-nrf/src/gpio.rs

```diff
@@ -205,6 +166,101 @@ impl<'d, T: Pin> StatefulOutputPin for Output<'d, T> {
     }
 }
 
+/// GPIO flexible pin.
+///
+/// This pin can either be a disconnected, input, or output pin.
+pub struct FlexPin<'d, T: Pin> {
+    pub(crate) pin: T,
+    phantom: PhantomData<&'d mut T>,
+}
+
+impl<'d, T: Pin> FlexPin<'d, T> {
+    /// Wrap the pin in a `FlexPin`.
+    ///
+    /// The pin remains disconnected.
+    pub fn new(pin: impl Unborrow<Target = T> + 'd) -> Self {
+        unborrow!(pin);
+        // Pin will be in disconnected state.
+        Self {
+            pin,
+            phantom: PhantomData,
+        }
+    }
+
+    pub fn to_input(&self, pull: Pull) {
+        self.pin.conf().reset(); // TODO is this necessary?
+        init_input(&self.pin, pull);
+    }
+
+    pub fn to_output(&self, initial_output: Level, drive: OutputDrive) {
```

Comment:
It might be nicer to not require initial_output here? It'd make switching faster since it only has to write one reg, The behavior would be keeping whatever output was last set.

new() could document the initial output state is unspecified, so you have to set_high or set_low before the first set_output

Reply:
Ok cool that sounds good.